### PR TITLE
[RENA-6249] Date Picker is not showing when clicking the calendar icon

### PR DESCRIPTION
**Pull request by mzbeetnoff**
## Resolves
https://moveinc.atlassian.net/browse/RENA-6249

## Why
The date picker UI no longer shows when clicking the calendar icon.

## Solution
Explicitly position the input date picker default icon behind the calendar icon.

| Page | Screenshot |
| --- | ------ |
| Canvas | image |
| Docs | image |

### CODE REVIEW

File: packages/controls/src/input.js [hunk 1]
```diff
@@ -115,6 +115,11 @@ export const Input = forwardRef(function Input(
               background: "transparent",
               opacity: 1,
               cursor: "pointer",
+              position: "absolute",
+              right: "2rem",
+              top: "3rem",
+              zIndex: 100,
+              fontSize: "24px",
```
Comment (mzbeetnoff) [inline packages/controls/src/input.js:122]: this is the size of the calendar icon, which is rendered in `px`.

**Review by stassribnyi — DISMISSED** [review]
LGTM!

File: packages/controls/src/input.js [hunk 2]
```diff
@@ -115,6 +115,11 @@ export const Input = forwardRef(function Input(
               background: "transparent",
               opacity: 1,
               cursor: "pointer",
+              position: "absolute",
+              right: "2rem",
+              top: "3rem",
+              zIndex: 100,
```
Comment (tpstar) [inline packages/controls/src/input.js:121]: @mzbeetnoff Is setting zIndex here necessary? I think you added zIndex to have the calendar-picker-indicator cover our icon, but it seems like it already covers our icon?

File: packages/controls/src/input.js [hunk 3]
```diff
@@ -133,6 +138,7 @@ export const Input = forwardRef(function Input(
               position: "absolute",
               right: "2rem",
               top: "3rem",
+              cursor: "pointer",
```
Comment (tpstar) [inline packages/controls/src/input.js:141]: @mzbeetnoff Since the calendar-picker-indicator already has the cursor, I don't think this is necessary?

**Review by tpstar — COMMENTED** [review]
@mzbeetnoff Nice job. This fix worked as expected.
A few nit-picks
1. Adding zIndex to the calendar-picker-indicator might have some unforeseen consequence and is not necessary? (e.g. If the date input is behind a dialog, and clicking a part of the dialog can activate the calendar?)
2. The cursor for our own calendar icon doesn't seem to be necessary.

**Comment by mzbeetnoff** [comment]
Thanks @tpstar for catching that, I updated the change 👍

**Review by tpstar — DISMISSED** [review]
@mzbeetnoff Thanks for addressing my comments. LGTM

**Review by jgrubard — COMMENTED** [review]
@mzbeetnoff Great find! I can confirm that I was running an outdated verision of Chrome locally, where the date picker worked as expected, and after updating I see the same symptom where clicking the calendar icon does nothing, and there appears to be a hidden/invisible icon right beside where you type the date the prompts the date picker when clicked.

On thing I would suggest -- could you update storybook to include a date input? I did this to test locally and confirm the issue is present on `release` and fixed on your branch. I think you can update the `Input` story in `packages/controls/stories/_controls.stories.mdx` with something like the following snippet:

```jsx
<Story name="Input">
  <Container as={Flex} sx={{ p: "2rem", flexDirection: "column" }}>
    <Box as="section">
      <Box as="h3" sx={{ mb: "1rem" }}>Text Input</Box>
      <Input icon={Search} label="Find Your Next Property" />
    </Box>
    <Box as="section">
      <Box as="h3" sx={{ mb: "1rem" }}>Date Input</Box>
      <Input type="date" label="Start Date" />
    </Box>
  </Container>
</Story>
```
Which gives us the following UI.
|Screenshot|
| - |
|Screen Shot 2024-03-12 at 1 56 05 PM|

One thing I _did_ notice, is that in Firefox, I can see the original calendar icon behind our custom icon, however this is not a blocker and it already exists -- I don't believe it was a result of any changes you made.

**Comment by mzbeetnoff** [comment]
@jgrubard yeah, it does exist before this change. I looked into this a bit more since your comment and I can't seem to find a solution to fix this in Firefox 😢

File: packages/controls/stories/_controls.stories.mdx [hunk 4]
```diff
@@ -63,6 +63,40 @@ Checkboxes are user selection tools when multiple options can be chosen at the s
...
+
+<Canvas>
+  <Story name="Date">
+    <Container>
+      <Box as="section">
+        <Input type="date" label="Start Date" />
+      </Box>
+    </Container>
+  </Story>
+</Canvas>
```
Comment (jgrubard) [inline packages/controls/stories/_controls.stories.mdx:76]: @mzbeetnoff Just nitpicking, I like the spacing for the `Canvas` but it makes the documentation and the provided code messy/unclear. Maybe we just simplify and update `Date` to `Date Input` to make sure it's easier to understand? Something like
```suggestion
## Date Input

<Canvas>
  <Story name="Date Input">
    <Input type="date" label="Start Date" />
  </Story>
</Canvas>
```

**Pull request merged**